### PR TITLE
Date and time/new date methods

### PR DESCRIPTION
**Pull request by lordstingray**
### Motivation / Background

I came across one particular check where it was required to validate if current date is also a start of the month or not. Which brings me to find any built-in methods and found nothing but there was one method `today?` which convinced me add these two methods as well.

### Detail

This pull request will add two different methods:
- `is_start_of_month?` with alias `is_beginning_of_month`.
- `is_end_of_month`, this method does not have any alias.

Both methods will help to verify if current date is also a start of the month or not, uses same approach for the second method as well to check if current date is also end of the month.  

### Additional information

You can use them like this:

```
Time.zone.today.is_start_of_month?
Time.zone.today.is_beginning_of_month?
Time.zone.today.is_end_of_month?

Date.current.is_start_of_month?
Date.current.is_beginning_of_month?
Date.current.is_end_of_month?
```

### Checklist

Before submitting the PR make sure the following are checked:

* [ ✔️  ] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [ ✔️ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ✔️ ] Tests are added or updated if you fix a bug or add a feature.
* [✔️  ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: activesupport/CHANGELOG.md [hunk 1]
```diff
@@ -530,4 +530,17 @@
 
     *Jean Boussier*
 
+*   New methods to check if current date is start of month or end of month, returns boolean.
```
Comment (zzak) [inline activesupport/CHANGELOG.md:533]:
```suggestion
* Add `DateAndTime#is_start_of_month?`, `DateAndTime#is_beginning_of_month?`, and `DateAndTime#is_end_of_month?` to check if current date is start of month or end of month, returns boolean.

```

I could be wrong about the class/module there, but we generally want to specify _which_ methods are being added in the changelog.

Also, there should be a new line between paragraphs.

File: activesupport/CHANGELOG.md [hunk 2]
```diff
@@ -530,4 +530,17 @@
...
+    ```
+    Time.zone.today.is_start_of_month?
+    Date.current.is_start_of_month?
+
+    Time.zone.today.is_beginning_of_month?
+    Date.current.is_beginning_of_month?
+
+    Time.zone.today.is_end_of_month?
+    Date.current.is_end_of_month?
+    ```
```
Comment (zzak) [inline activesupport/CHANGELOG.md:543]:
```suggestion
    ```

```

New line :bow:

**Comment by lordstingray** [comment]
@zzak updated PR after your suggested changes 👍🏻

**Comment by rafaelfranca** [comment]
Thank you for the pull request. I considered this PR carefully and I prefer to not add those methods. Why only beginning of month if we have methods for beginning of year and beginning of quarter. If we add methods for all those periods they aren't going to be used. If we don't add we will always ask ourselves why we have for one period but not others.

So I prefer to not have any of them given it isn't hard to implement.

**Pull request closed without merging**